[Term Entry] Java Calendar internalGet() (#3102) (#3192)
Codecademy/docs: content/java/concepts/calendar/terms/internalGet/internalGet.md

---
Title: '.internalGet()'

> Description: 'Returns the value of a given field'

Reviewer: ```suggestion
Description: 'Returns the value of a given field.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Calendar'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> The **`.internalGet()`** method is a protected method used to retrieve the value of a field from a Calendar instance. In other words it takes a field as input and returns its corresponding value. This method is generally used within the Calendar class or its sub classes.

Reviewer: ```suggestion
The **`.internalGet()`** method is a protected method used to retrieve the value of a field from a `Calendar` instance. In other words, it takes a field as input and returns its corresponding value. This method is generally used within the `Calendar` class or its sub classes.
```

## Syntax

```pseudo

> protected final int internalGet(int field)

Reviewer: ```suggestion
myCalendar.internalGet(int field)
```

```

> 

Reviewer: There should be a note here after the syntax regarding the parameters taken, return value, or any other valuable information

> ## Example 1

Reviewer: ```suggestion
## Example 
```

> In this example, the `InternalGetExample` class inherits the `GregorianCalendar` class which is a concrete subclass of the Calendar class. The code first instantiates a new object of type `InternalGetExample` with the current date which uses the Gregorian calendar system.

Reviewer: ```suggestion
In this example, the `InternalGetExample` class inherits the `GregorianCalendar` class. The code first instantiates a new object of type `InternalGetExample` with the current date which uses the Gregorian calendar system.
```

> **Note:** There are multiple calendar systems that `Calendar.getInstance()` provides depending on the locale. These include `GregorianCalendar`, `BuddhistCalendar`, and `JapaneseImperialCalendar`.

```java

> import java.util.GregorianCalendar;

Reviewer: update the code to use 2 spaces for indentation instead of tabs
